### PR TITLE
Add new Filter Parameter Widget to the code generators

### PR DESCRIPTION
**Pull request by imikejackson**
Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

### CODE REVIEW

File: Source/DevHelper/CodeGenerators/LinkedPathCreationWidgetCodeGenerator.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+LinkedPathCreationWidgetCodeGenerator::~LinkedPathCreationWidgetCodeGenerator() = default;
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+QString LinkedPathCreationWidgetCodeGenerator::generateSetupFilterParameters()
+{
+  QString s;
+  QTextStream out(&s);
+  out << "  parameters.push_back(SIMPL_NEW_AM_WITH_LINKED_DC_FP(\"" << getHumanLabel() << "\", " << getPropertyName() << ", " << getCategory() << ", @FilterName@));";
```
Comment (mmarineBlueQuartz) [inline Source/DevHelper/CodeGenerators/LinkedPathCreationWidgetCodeGenerator.cpp]: This is should not compile.  LinkedPathCreationFilterParameter requires an additional parameter for AttributeMatrices (the one used here), and yet another one for DataArrays (not available through this implementation).

**Review by mmarineBlueQuartz — CHANGES_REQUESTED** [review]
The LinkedPathCreationFilterParameter macro generated will not compile, and only the AttributeMatrix version is used.  This generator will not call the DataArray constructor.

The LinkedPathCreationFilterParameter also works with a single QString property, not DataArrayPaths.

File: Source/DevHelper/CodeGenerators/LinkedPathCreationWidgetCodeGenerator.cpp (hunk 1 above)
Comment (imikejackson) [reply Source/DevHelper/CodeGenerators/LinkedPathCreationWidgetCodeGenerator.cpp]: Updated generated code to induce a compiler error after the code is generated.

File: Source/DevHelper/CodeGenerators/LinkedPathCreationWidgetCodeGenerator.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+QString LinkedPathCreationWidgetCodeGenerator::generateFilterParameters()
+{
+  QString contents;
+  QTextStream ss(&contents);
+  ss << "    SIMPL_FILTER_PARAMETER(DataArrayPath, " + getPropertyName() + ")\n";
+  ss << "    Q_PROPERTY(DataArrayPath " + getPropertyName() + " READ get" + getPropertyName() + " WRITE set" + getPropertyName() + ")";
```
Comment (mmarineBlueQuartz) [inline Source/DevHelper/CodeGenerators/LinkedPathCreationWidgetCodeGenerator.cpp]: Change this to a QString property and update the copyright header to match the .h file, and this should be good.

**Review by mmarineBlueQuartz — APPROVED** [review]
This looks good.

**Pull request merged**